### PR TITLE
feat(config): Introduce ephemeral and persistent storage [fixes NET-759 NET-760]

### PR DESCRIPTION
**Pull request by gurinderu**
## Description
Introduce ephemeral and persistent storage to Nox.

## Motivation
The goal is to separate data stored in a ~10Gb RAM Disk (aka ephemeral storage) from data stored on SSD but limited to 100-500Mb.

## Related Issue(s)
[NET-759](https://linear.app/fluence/issue/NET-759/introduce-ephemeral-and-persistent-storage-to-nox-config)
[NET-760](https://linear.app/fluence/issue/NET-760/insert-tmp-and-storage-into-every-moduleconfig)

## Additional Notes
- Fixed core manager persistence
- Fixed worker runtime shutdown
- Exposed the applied config in the CreatedSwarm for tests

## Checklist
- [ ] The code follows the project's coding conventions and style guidelines.
- [ ] All tests related to the changes have passed successfully.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing unit tests have passed.
- [ ] I have self-reviewed my code and ensured its quality.
- [ ] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Code has been reviewed for quality and adherence to [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: crates/nox-tests/tests/modules.rs [hunk 1]
```diff
@@ -88,7 +88,6 @@ async fn test_add_module_mounted_binaries_forbidden() {
         "logger_enabled": true,
         "wasi": {
             "envs": json!({}),
-            "preopened_files": vec!["/tmp"],
```
Comment (ValeryAntopol) [inline crates/nox-tests/tests/modules.rs:91]: rollback it

File: crates/service-modules/src/modules/fixture.rs [hunk 2]
```diff
@@ -34,7 +34,6 @@ pub fn module_config(import_name: &str) -> JValue {
         "logger_enabled": true,
         "wasi": {
             "envs": json!({}),
-            "preopened_files": vec!["/tmp"],
```
Comment (ValeryAntopol) [inline crates/service-modules/src/modules/fixture.rs:37]: rollback it

File: particle-builtins/src/builtins.rs [hunk 3]
```diff
@@ -1020,7 +1020,6 @@ fn make_module_config(args: Args) -> Result<JValue, JError> {
     let _max_heap_size: Option<String> = Args::next_opt("max_heap_size", &mut args)?;
 
     let logger_enabled = Args::next_opt("logger_enabled", &mut args)?;
-    let preopened_files = Args::next_opt("preopened_files", &mut args)?;
```
Comment (ValeryAntopol) [inline particle-builtins/src/builtins.rs:1023]: rollback and leave a comment like for couple of lines above

File: crates/nox-tests/tests/modules.rs (hunk 1 above)
Comment (gurinderu) [reply crates/nox-tests/tests/modules.rs:91]: done

**Pull request merged**